react/components/atoms/switcher/switcher.js

```diff
@@ -0,0 +1,160 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, Text, Animated, View, Platform } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles } from "../../../util";
+
+import { Touchable } from "../touchable";
+
+export class Switcher extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            checked: PropTypes.bool,
+            disabled: PropTypes.bool,
+            checkedText: PropTypes.string,
+            uncheckedText: PropTypes.string,
+            variant: PropTypes.string,
+            animationDuration: PropTypes.number,
+            marginLeftValue: PropTypes.number,
+            colorInputRangeValue: PropTypes.number,
+            onValueUpdate: PropTypes.func
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            checked: false,
+            disabled: false,
+            variant: "colored",
+            checkedText: null,
+            uncheckedText: null,
+            animationDuration: 200,
+            marginLeftValue: 20,
+            colorInputRangeValue: 150,
+            onValueUpdate: () => {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            checkedData: this.props.checked,
+            marginLeft: this.props.checked
+                ? new Animated.Value(this.props.marginLeftValue)
+                : new Animated.Value(0),
+            backgroundColor: this.props.checked
+                ? new Animated.Value(this.props.colorInputRangeValue)
+                : new Animated.Value(0)
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.checked !== this.props.checked) {
+            this.setChecked(this.props.checked);
+        }
+    }
+
+    _getText() {
+        return this.state.checkedData ? this.props.checkedText : this.props.uncheckedText;
+    }
+
+    _switcherStyle = () => {
+        const isColored = this.props.variant === "colored";
+        return [
+            styles.switcher,
+            {
+                backgroundColor: this.state.backgroundColor.interpolate({
+                    inputRange: [0, 150],
+                    outputRange: ["#cccccc", isColored ? "#507bf8" : "#1d1d1d"]
+                }),
+                borderColor: this.state.backgroundColor.interpolate({
+                    inputRange: [0, 150],
+                    outputRange: ["#cccccc", isColored ? "#507bf8" : "#1d1d1d"]
+                }),
+                opacity: this.props.disabled ? 0.3 : 1
+            }
+        ];
+    };
+
+    _buttonStyle = () => {
+        return [styles.button, { marginLeft: this.state.marginLeft }];
+    };
+
+    setChecked = checked => {

```

Comment:
Public methods (as in "methods that might be called from other component" = "methods without underscore") go right after "componentDidUpdate". Order:
- Props
- Constructor
- Component lifecycle hooks
- Private methods(_stuff)
- Event handlers
- Render methods